### PR TITLE
Update deps for Ledger, Consensus, et al

### PR DESCRIPTION
**Pull request by nfrisby**
This PR does two things.

- It updates many dependencies.

- The necessary integration is just trivial refactoring. Beyond that, this PR also updates the log representation of some tracer events; typo fixes and a new shape for ChainSync server events.

```
           cardano-ledger f31c29add - Merge pull request #2819 from input-output-hk/andre/babbage (6 days ago) <Alexey Kuleshevich>
iohk-monitoring-framework 066f700   - Merge #627 (5 days ago) <iohk-bors[bot]>
                   io-sim f4183f2   - Merge pull request #2 from input-output-hk/coot/check-stylish-script (12 days ago) <Marcin Szamotulski>
        ouroboros-network c254d02b9 - Merge #3766 (5 hours ago) <iohk-bors[bot]>
                   plutus d24a7540e - Backport: PLT-250: mkEvaluationContext now lives in MonadError CostModelApplyError (#4641) (9 days ago) <Nikolaos Bezirgiannis>
          typed-protocols 181601b   - Merge pull request #4 from input-output-hk/coot/check-stylish (12 days ago) <Marcin Szamotulski>
```

### CODE REVIEW

**Comment by nfrisby** [comment]
This is Draft because it points at the commit of https://github.com/input-output-hk/ouroboros-network/pull/3774 instead of the merged result of it.

https://github.com/input-output-hk/ouroboros-network/pull/3774 has not yet merged, because it similarly points at https://github.com/input-output-hk/iohk-monitoring-framework/pull/627.

https://github.com/input-output-hk/iohk-monitoring-framework/pull/627 has not yet merged, because `bors merge` has time out four or fives times now :'(

**Review by lehins — APPROVED** [review]
LGTM

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs
Comment (michaelpj) [inline cardano-api/src/Cardano/Api/ProtocolParameters.hs:756]: do you maybe want to propagate the error?

File: cardano-api/gen/Gen/Cardano/Api.hs
Comment (michaelpj) [inline cardano-api/gen/Gen/Cardano/Api.hs:95]: FYI these errors have a `Pretty` instance if you want to use that.

**Review by michaelpj — APPROVED** [review]

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs
Comment (nfrisby) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:756]: This is a fair question. My goal was this PR was to unblock the dependency bumps.

cc @Jimbo4350 re propagating this error now?

**Comment by nfrisby** [comment]
I have now marked this Ready for Review and will notify @Jimbo4350 and @kosyrevSerge

**Review by amesgen — APPROVED** [review]

File: cardano-api/gen/Gen/Cardano/Api.hs
Comment (Jimbo4350) [reply cardano-api/gen/Gen/Cardano/Api.hs:95]: Yep that sounds good

File: cardano-api/src/Cardano/Api/ProtocolParameters.hs
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/ProtocolParameters.hs:756]: Yeah we may as well do it now. I can take over.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs:1029]: Where can I find more information about the "horizon"? I think we can improve this error message.

**Review by Jimbo4350 — COMMENTED** [review]
LGTM. I can take over and make some adjustments.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs
Comment (nfrisby) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs:1029]: 1) I'm unaware of user facing documentation for this. 😬 

2) There's a whole chapter of the Consensus Report about it "Part IV Hard Fork Combinator", "Chapter 17 Time", where "beyond the horizon" means "past the Forecast range". But it's very abstract/fundamental/not necessarily in "Cardano terminology".

3) High-level summary (@JaredCorduan please review this):

* the horizon is either the end of the current epoch or the end of the next epoch
    * it's the end of the current epoch if the ledger state does not determine whether the current epoch is the end of the current era
    * it's the end of the next epoch if the ledger state does determine whether the current epoch is the end of the current era
    * (ie the horizon is the end of the current epoch if and only if we don't know whether it's the last epoch of this era)
    
* when does the ledger state does determine whether the current epoch is the end of the current era?

    * the transition happens at the end of the current epoch if and only if all the necessary proposal/voting transactions are in a block on the chain too deep to be rolled back _and_ there's a stability window's worth of slots between that block and the end of the current epoch.
    * The current epoch is determined as the last of its era when the above is already true in the ledger state.
    * The current epoch is determined as _not_ the last of its era when the above _cannot_ be true, eg it's not true now and we're already within a stability window of the end of the epoch.
    * Otherwise, it's not determined: ie it's not true now but it still might become true.

Comment (JaredCorduan) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs:1029]: That all sounds correct to me, though I might re-phrase it a bit. I would say that the ledger is ignorant about transitions to future eras, and that the consensus layer uses the ledger state to determine if an era is ending.

In particular, votes to change the protocol parameters in the next epoch _must_ occur within two stability windows of the end of the epoch (the vote states what epoch it is intended for, you can also place a pending vote for the following epoch). After one stability window from the end of the epoch (so one stability window _after_ voting has ended), the consensus layer looks to see if the major protocol version (which is a protocol parameter). is going to change, and hence usher in a new era (provided it is not an intra-era hard fork).

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs:1029]: I will come back to this another time, thanks for the input

**Comment by nfrisby** [comment]
bors retry

**Comment by amesgen** [comment]
FTR: I asked for code owner approvals on [Slack](https://input-output-rnd.slack.com/archives/CR599HMFX/p1654248700006379)

**Review by JaredCorduan — APPROVED** [review]
thank you @nfrisby !

**Comment by Jimbo4350** [comment]
bors merge

**Review by deepfire — APPROVED** [review]

**Comment by nfrisby** [comment]
bors retry

**Pull request merged**
